Mathlib/Logic/Function/OfArity.lean

```diff
@@ -26,49 +26,56 @@ namespace Function
 
 Note that this is not universe polymorphic, as this would require that when `n=0` we produce either
 `Unit → β` or `ULift β`. -/
-def OfArity (α β : Type u) : ℕ → Type u
-  | 0 => β
-  | n + 1 => α → OfArity α β n
+def OfArity (α β : Type u) (n : ℕ) : Type u := OfHArity (fun (_ : Fin n) => α) β
```

Comment:
I wonder if we can get away with making this an `abbrev`; does anything break?

Reply:
Seems like it's fine, the ZFC.Basic file doesn't break in any new way